k8s/scripts/coredns_patch.sh

```diff
@@ -45,11 +47,17 @@ else
     cat $PATCH
     echo '=== end patched Corefile listing ==='
 
-    kubectl get cm coredns -n kube-system -o json | jq --arg replace "`cat $PATCH`" '.data.Corefile = $replace' | kubectl apply -f -
+    jq --arg replace "`cat $PATCH`" '.data.Corefile = $replace' $COREDNS_CM | kubectl apply -f -
 fi
 echo 'Running kubectl rollout restart for coredns...'
 kubectl rollout restart deployment/coredns -n kube-system
 echo 'Verifying that DNS resolution works inside the cluster'
 DNS_CHECK_SCRIPT="for svc in ${SERVICES[*]}; do echo Checking DNS for \$svc...; host \$svc.$DOMAIN | grep \$svc.$DOMAIN; done"
```

Comment:
instead of greping for the output of the host command, can we use the exit status?
Also, I suppose we have covered the negative test cases for resolution: all success, one erroneous entry

Reply:
interesting finding about the exit status from host, on my local machine this is what I get:
```
$ host web1.local.trustbloc.dev || echo $?
Host web1.local.trustbloc.dev not found: 3(NXDOMAIN)
1
```

Reply:
I guess it might be smth specific to the specific `nslookup` or `host` command binary, but here are my results:
```
DOMAIN="" kubectl run dnsutils --image=gcr.io/kubernetes-e2e-test-images/dnsutils:1.3 --rm --attach --command --restart=Never -- sh -ec "for s in ${SERVICES[*]}; do nslookup \$s.$DOMAIN; done"
If you don't see a command prompt, try pressing enter.
Address:	10.96.0.10#53

Non-authoritative answer:
*** Can't find orb-1: No answer

Server:		10.96.0.10
Address:	10.96.0.10#53

Non-authoritative answer:
*** Can't find orb-kms: No answer

Server:		10.96.0.10
Address:	10.96.0.10#53

Non-authoritative answer:
*** Can't find vct: No answer
...
```
So, the **dnsutils** container `nslookup` command returns exit code 0 and displays the "no answer" result, while in fact, these domains are not resolvable. 
`hosts` command behaves in a similar fashion. 
Also, I could not get any reliable results from using the **busybox** container `nslookup` command.

Reply:
I might be wrong, but the case above looks different; empty domain var.
running this:
kubectl run -it dnsutils  --image=gcr.io/kubernetes-e2e-test-images/dnsutils:1.3 --rm --attach --command --restart=Never

/ # host web1.local.trustbloc.dev || echo $?
Host web1.local.trustbloc.dev not found: 3(NXDOMAIN)
1

nslookup web1.local.trustbloc.dev || echo $?
Server:		10.96.0.10
Address:	10.96.0.10#53

** server can't find web1.local.trustbloc.dev: NXDOMAIN

1

Reply:
Yes, but the empty domain test case above represents a test case of the nonexistent domain and is an example when hosts / nslookup return exit status 0 for the nonexistent domain.

Reply:
- the target test case is narrow enough: test coredns host-file plugin resolution after the patch, and its not general dns resolution.
- non-empty DOMAIN is a precondition, so it should out of scope for the test.
- host return 0|1 for the above.

Reply:
OK, so to me the exit code check just seems to be less reliable than the "grep" check (because the "grep" check covers more potential problems). 
We can use the exit code check instead of the grep check if you consider it good enough for this case.

Reply:
@alialkhalidi also please consider this example run (WITH the DOMAIN properly set):
```
$ kubectl run dnsutils --image=gcr.io/kubernetes-e2e-test-images/dnsutils:1.3 --rm --attach --command --restart=Never -- sh -ec "set -e; for s in ${SERVICES[*]} aaaaa bbbbb; do nslookup \$s.local.trustbloc.dev; done"
Server:		10.96.0.10
Address:	10.96.0.10#53

Name:	adapter-hydra.local.trustbloc.dev
Address: 192.168.49.2
....
SKIPPING SUCCESSFUL DNS RESOLVES FROM THE OUTPUT
....

Name:	static-file-server.local.trustbloc.dev
Address: 192.168.49.2

Server:		10.96.0.10
Address:	10.96.0.10#53

*** Can't find aaaaa.local.trustbloc.dev: No answer

Server:		10.96.0.10
Address:	10.96.0.10#53

** server can't find bbbbb.local.trustbloc.dev: NXDOMAIN

pod "dnsutils" deleted
pod default/dnsutils terminated (Error)
```

Note that it did not fail on the first `aaaaa.local.trustbloc.dev`, but failed on the second `bbbbb.local.trustbloc.dev`
THe exit code check just seems inconsistent in such case

Reply:
I've tested the exit code several times on GH cloud worker and it seems to be reliable. 
Pushed a change to use the exit code from the `host` command.